libs/actions/sendmail.c

```diff
@@ -0,0 +1,226 @@
+/*
+	Copyright (C) 2015 CurlyMo & Niek
+
+	This file is part of pilight.
+
+	pilight is free software: you can redistribute it and/or modify it under the
+	terms of the GNU General Public License as published by the Free Software
+	Foundation, either version 3 of the License, or (at your option) any later
+	version.
+
+	pilight is distributed in the hope that it will be useful, but WITHOUT ANY
+	WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+	A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with pilight. If not, see	<http://www.gnu.org/licenses/>
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <fcntl.h>
+
+#include "action.h"
+#include "log.h"
+#include "dso.h"
+#include "json.h"
+#include "common.h"
+#include "config.h"
+#include "pilight.h"
+#include "mail.h"
+#include "sendmail.h"
+#include "settings.h"
+
+
+//check arguments and settings
+static int actionSendmailArguments(struct JsonNode *arguments) {
+	struct JsonNode *jsubject = NULL;
+	struct JsonNode *jmessage = NULL;
+	struct JsonNode *jto = NULL;
+	struct JsonNode *jvalues = NULL;
+	struct JsonNode *jchild = NULL;
+	struct JsonNode *temp = NULL;
+	char *stmp = NULL;
+	int nrvalues = 0, itmp = 0;
+	
+	jsubject = json_find_member(arguments, "SUBJECT");
+	jmessage = json_find_member(arguments, "MESSAGE");
```

Comment:
Now I understand, but I do find them usefull.

In my own case, all messages sent from rules must go to the same 
recipient (myself), with always  the same sender (pilight) and most of 
them with the same subject (pilight event). In those rules I only need 
to use the "MESSAGE" argument , because the others are all in the settings.

But if you don't like these setting let's throw them out.

CurlyMoo schreef op 17-3-2015 om 20:58:
>
> In libs/actions/sendmail.c 
> <https://github.com/pilight/pilight/pull/155#discussion_r26609969>:
>
> > +#include "settings.h"
> > +
> > +
> > +//check arguments and settings
> > +static int actionSendmailArguments(struct JsonNode *arguments) {
> > +	struct JsonNode *jsubject = NULL;
> > +	struct JsonNode *jmessage = NULL;
> > +	struct JsonNode *jto = NULL;
> > +	struct JsonNode *jvalues = NULL;
> > +	struct JsonNode *jchild = NULL;
> > +	struct JsonNode *temp = NULL;
> > +	char *stmp = NULL;
> > +	int nrvalues = 0, itmp = 0;
> > +	
> > +	jsubject = json_find_member(arguments, "SUBJECT");
> > +	jmessage = json_find_member(arguments, "MESSAGE");
>
> I was missing the *not* :)
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/pilight/pilight/pull/155/files#r26609969>.
>